src/compressed_tensors/compressors/base.py

```diff
@@ -169,6 +170,10 @@ def decompress_module(self, module: Module):
         :param module: PyTorch module to decompress
         :return: tensor of the decompressed weight, or None if module is not quantized
         """
+
+        params_device = next(module.parameters()).device
+        device = "cpu" if has_offloaded_params(module) else params_device
```

Comment:
This is technically safe because (afaict) `decompress_module` is only used by `CompressedLinear`, and no operations happen between it being returned by this function and the weight being registered via `register_offload_parameter`.

I'm good to keep this, just something to be aware of

Reply:
yeah the alternate would be to register it here but I think the general pattern is that the decompressor just returns the decompressed value followed by a call to register/update